File: udgaard/src/main/java/com/skrymer/udgaard/model/Ovtlyr9EntryStrategy.java
package com.skrymer.udgaard.model;

/**
 * Entry strategy based on the ovtlyr 9 criteria:
 * 
 * Market 
 *  Market breadth is in an uptrend
 *  Spy is in an uptrend OK
 *  Spy has a buy signal OK
 * 
 * Sector:
 *  Sector breadth is in an uptrend  OK
 *  Sector is getting greeder OK
 * 
 * Stock: 
 *  is in an uptrend OK
 *  has a buy signal within the last 2 days OK
 *  is getting more greedy OK 
 */
public class Ovtlyr9EntryStrategy implements EntryStrategy {

    @Override
    public boolean test(StockQuote quote) {
        return 
            // The stock is in an uptrend
            quote.isInUptrend() && 
            // We have a buy signal
            quote.getLastBuySignal() != null &&
            // The buy signal is within the last 2 days of the quote
            quote.getLastBuySignal().isAfter(quote.getDate().minusDays(3)) && 
            // The stock heatmap is getting greeder
            quote.isGettingGreeder() && 
            // The sector is in an uptrend
            quote.sectorIsInUptrend() && 
            // The sector heatmap is getting greeder 
            quote.sectorIsGettingGreeder() && 
            // Spy has a buy signal
            quote.hasSpyBuySignal() &&
            // Spy is in an uptrend
            quote.isSpyInUptrend() &&
            // Market is in an uptrend
            quote.marketIsInUptrend();
    }

}

File: new_udgaard/udgaard/src/main/kotlin/com/skrymer/udgaard/model/strategy/Ovtlyr9EntryStrategy.kt
package com.skrymer.udgaard.model.strategy

import com.skrymer.udgaard.model.StockQuote

/**
 * Entry strategy based on the ovtlyr 9 criteria:
 *
 * Market
 * Market breadth is in an uptrend
 * Spy is in an uptrend
 * Spy has a buy signal
 *
 * Sector:
 * Sector breadth is in an uptrend
 * Sector is getting greeder
 *
 * Stock:
 * is in an uptrend
 * has a buy signal within the last 2 days
 * is getting more greedy
 * close price is over the 10EMA
 */
class Ovtlyr9EntryStrategy : EntryStrategy {
    override fun test(quote: StockQuote?): Boolean {
        return StockIsInUptrend()
            .and(ClosePriceOver10Ema())
            .and(CurrentBuySignal())
            .and(StockGettingGreeder())
            .and(SectorIsInUptrend())
            .and(SectorIsGettingGreeder())
            .and(HasSpyBuySignal())
            .and(SpyIsInUptrend())
            .and(MarketIsInUptrend())
            .test(quote)
    }

    override fun description(): String {
        return "Ovtlyr 9 entry strategy"
    }
}
